### DISCUSSION TITLE
How does one specify a dependency with a dash (or any non-canonical character) in its name?

### DISCUSSION THREAD

**Discussion by dan-fritchman** [python-poetry/poetry#7062] [General] [answered]
Example:  
https://github.com/dan-fritchman/UseScikitLearn

This is a result of running `poetry new`, and then attempting to add the popular package `scikit-learn`:  
https://pypi.org/project/scikit-learn/

Attempting to install generates: 

```
$ poetry --version
Poetry version 1.1.12
$ poetry install

  TypeError

  expected string or bytes-like object

  at .../site-packages/poetry/core/utils/helpers.py:27 in canonicalize_name
       23│ _canonicalize_regex = re.compile(r"[-_]+")
       24│ 
       25│ 
       26│ def canonicalize_name(name):  # type: (str) -> str
    →  27│     return _canonicalize_regex.sub("-", name).lower()
       28│ 
       29│ 
       30│ def module_name(name):  # type: (str) -> str
       31│     return canonicalize_name(name).replace(".", "_").replace("-", "_")
```

Presumably it's the `-` which gets in the way, and fails in this `canonicalize`-ing step. 

So, how does one do that?  
Either in `pyproject.toml`, or via `poetry add`?  
Wrapping it in quotes doesn't seem to work (with either). Nor do I see a "long package dependency" format which includes a string-valued name in here: 
https://github.com/python-poetry/poetry-core/blob/main/src/poetry/core/json/schemas/poetry-schema.json

Thanks!

**Comment by neersighted** [comment, marked as answer]
This is a solved bug and you're on an ancient version of Poetry. Please update to a supported version (e.g. the 1.2 branch) and try again.

```sh
$ docker run --rm -it --entrypoint bash python:3.10
# pip install poetry==1.2.2
# poetry new repro && cd repro
# poetry add scikit-learn
[...]
  • Installing numpy (1.23.4)
  • Installing joblib (1.2.0)
  • Installing scipy (1.9.3)
  • Installing threadpoolctl (3.1.0)
  • Installing scikit-learn (1.1.3)
```

**Comment by dan-fritchman** [comment]
Alright, yep, that's on me. 

My mistake: that poetry-version was coming from `conda`: 
https://anaconda.org/conda-forge/poetry
...and it really ain't being updated. 

Thanks!
